modules/opensearch/opensearch.module

```diff
@@ -0,0 +1,432 @@
+    ),
+  );
+
+  $form['opensearch']['authentification']['opensearch_auth_name'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Name'),
+    '#default_value' => variable_get('opensearch_auth_name', ''),
+    '#states' => $auth_states,
+  );
+
+  $form['opensearch']['authentification']['opensearch_auth_pass'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Password'),
+    '#default_value' => variable_get('opensearch_auth_pass', ''),
+    '#states' => $auth_states,
+  );
+
+  $form['opensearch']['authentification']['opensearch_auth_group'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Group'),
+    '#default_value' => variable_get('opensearch_auth_group', ''),
+    '#states' => $auth_states,
+  );
+
+  $form['opensearch']['opensearch_recommendation_url'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Recommendation service URL'),
+    '#description' => t('URL to the recommendation webservice (Andre der har lånt...) , e.g. http://openadhl.addi.dk/1.1/'),
+    '#required' => FALSE,
+    '#default_value' => variable_get('opensearch_recommendation_url', ''),
+  );
+
+  $form['opensearch']['opensearch_enable_logging'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Enable logging'),
+    '#default_value' => variable_get('opensearch_enable_logging', FALSE),
+    '#weight' => 1,
+  );
+
+  $intervals = array(
+    60,
+    300,
+    900,
+    1800,
+    3600,
+    10800,
+    43200,
+    86400,
+    259200,
+    604800,
+  );
+
+  $options = array();
+  foreach ($intervals as $interval) {
+    $options[$interval] = format_interval($interval, 2);
+  }
+
+  $form['opensearch']['opensearch_cache_lifetime'] = array(
+    '#type' => 'select',
+    '#title' => t('Cache lifetime'),
+    '#default_value' => variable_get('opensearch_cache_lifetime', OPENSEARCH_DEFAULT_CACHE_LIFETIME),
+    '#options' => $options,
+    '#description' => t("Cache answers from the datawell for this time. At least a few minutes is recommended, so the same objects doesn't have to be fetched on each page request."),
+    '#weight' => 1,
+  );
+
+  $form['opensearch']['opensearch_search_profile'] = array(
+    '#type' => 'textfield',
+    '#title' => t('Search profile'),
+    '#description' => t('Profile name, used for filtering the search based on settings from vip.dbc.dk.'),
+    '#default_value' => variable_get('opensearch_search_profile', ''),
+  );
+  return system_settings_form($form);
+}
+
+/**
+ * Validation handler for opensearch_search_autocomplete_admin
+ */
+function opensearch_search_autocomplete_admin_validate($form, &$form_state) {
+  $form_state['values']['opensearch_search_autocomplete_suggestion_url'] = trim
+  ($form_state['values']['opensearch_search_autocomplete_suggestion_url']);
+}
+
+/**
+ * Get settings, and merge with defaults.
+ * @return array
+ */
+function opensearch_search_autocomplete_settings() {
+  $settings = array();
+
+  $base_settings['index'] = NULL;
```

Comment:
This has been transferred 1-1 and actually there is no reason to define a variable as an array before assigning values. [It will not cause notices](https://3v4l.org/nfFso) or the like.

I have added variable initialization.

Reply:
👍